include/benchmark/benchmark.h

```diff
@@ -180,17 +185,71 @@ void UseRealTime();
 namespace internal {
 class Benchmark;
 class BenchmarkFamilies;
+
+void StartBenchmarkTiming();
+void StopBenchmarkTiming();
 }
 
 // State is passed to a running Benchmark and contains state for the
 // benchmark to use.
 class State {
- public:
+public:
+  State(int max_iters, bool has_x, int x, bool has_y, int y, int thread_i)
+    : started_(false), total_iterations_(0), max_iterations_(max_iters),
+      has_range_x_(has_x), range_x_(x),
+      has_range_y_(has_y), range_y_(y),
+      bytes_processed_(0), items_processed_(0),
+      thread_index(thread_i)
+  {}
+
   // Returns true iff the benchmark should continue through another iteration.
-  bool KeepRunning();
+  BENCHMARK_ALWAYS_INLINE
+  bool KeepRunning() {
```

Comment:
can you please try this same change but with the implementation kept private? i'll be amazed if the overhead of the function call matches the inlined version, and the benefit to moving it is a simpler API with fewer internals exposed.